app/models/package_manager/maven/maven_central.rb

```diff
@@ -20,4 +20,20 @@ def self.one_version(raw_project, version_string)
   def self.missing_version_remover
     PackageManager::Base::MissingVersionRemover
   end
+
+  # maven-metadata.xml for Maven Central does not appear to be guaranteed to contain all relevant versions for a package
+  # So instead, if needed, we will retrieve the versions from the raw HTML index page
+  def self.versions(raw_project, name)
+    if raw_project && raw_project[:versions]
+      raw_project[:versions]
+    else
+      retrieve_versions(versions_from_html(name), name)
+    end
+  end
+
+  def self.versions_from_html(name)
+    get_html(MavenUrl.from_name(name, repository_base, NAME_DELIMITER).base).css("a").filter_map do |a|
```

Comment:
Should we navigate into the folder & persist the timestamp from an artifact?

Reply:
Additional thought: we shouldn't need to persist a timestamp, because in `retrieve_versions` we will still be collecting `.pom` files which will have the info we need for each version we collect.

Reply:
Isn't the published_at an important data point for versions though?

Reply:
Yeah I think you're correct, but we are getting all the same data that we previously would at a later step in the process, from the `.pom`s. This change in the process does not interfere with that.

https://github.com/librariesio/libraries.io/pull/3243/files#diff-a54c889fd056c7288db14f084d2ef1a242e3a96fa6d5b8ee6daa6116673f2615R30

https://github.com/librariesio/libraries.io/blob/main/app/models/package_manager/maven.rb#L189-L199

https://github.com/librariesio/libraries.io/blob/main/app/models/package_manager/maven.rb#L170